src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        let newUrl = this.win.location.href;
+        if (incomingFragment == "") {
+          if (newUrl.indexOf('#') == -1) {
+            newUrl = newUrl + '#.' + outgoingFragment;
+          } else {
+            newUrl = newUrl.replace('#', '#.' + outgoingFragment + '&');
+          }
+        } else {
+          newUrl = newUrl.replace(incomingFragment, outgoingFragment);
+        }
+        this.win.history.replaceState({}, '', newUrl);
```

Comment:
According to [this](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace)
> after using replace() the current page will not be saved in session History, meaning the user won't be able to use the back button to navigate to it.
I still want to overwrite the history to I guess `history.replaceState` is the one I want.